connectors/sources/dropbox.py

```diff
@@ -844,11 +857,35 @@ async def close(self):
         await self.dropbox_client.close()
 
     async def ping(self):
-        endpoint = EndpointName.PING.value
-        if self._dls_enabled():
+        try:
             endpoint = EndpointName.AUTHENTICATED_ADMIN.value
-        await self.dropbox_client.ping(endpoint=endpoint)
-        self._logger.info("Successfully connected to Dropbox")
+            await self.dropbox_client.ping(endpoint=endpoint)
+            self._logger.info("Successfully connected to Dropbox")
+        except ClientResponseError as exception:
+            if str(exception.request_info.url) == AUTHENTICATED_ADMIN_URL:
+                endpoint = EndpointName.PING.value
+                await self.dropbox_client.ping(endpoint=endpoint)
+                self._logger.info("Successfully connected to Dropbox")
+            else:
+                raise
+
+    async def set_user_info(self):
+        try:
+            endpoint = EndpointName.AUTHENTICATED_ADMIN.value
+            response = await self.dropbox_client.ping(endpoint=endpoint)
+        except ClientResponseError as exception:
+            if str(exception.request_info.url) == AUTHENTICATED_ADMIN_URL:
+                endpoint = EndpointName.PING.value
+                response = await self.dropbox_client.ping(endpoint=endpoint)
+            else:
+                raise
```

Comment:
Same as previous question.

Reply:
Also, if the first request returns a `ClientResponseError`, `response` will become the result from the ping endpoint instead of the authenticated admin endpoint. Will that not cause issues when trying to extract values for `self.dropbox_client.member_id` and `self.dropbox_client.root_namespace_id`?